### PR TITLE
Add soft capping to reversible embedding layer

### PR DESCRIPTION
**Pull request by mattdangerw**
Forgetting the final output soft-cap is a really easy mistake, and worse, outputs will still look plausible for generations without the softcap, just with worse actual results.

Adding this to our reversible embedding layer will be much more robust. As long as you use the layer to compute logits over the vocab, you can no longer forget the soft-cap.

### CODE REVIEW

**Review by SamanehSaadat — APPROVED** [review]
Thanks, Matt!
